docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+- Configuration, not unlike that provided in core data, will specify what message endpoints the metrics and CPE messages should be sent.
+  - Proposed configuration for each service for metric and CPE endpoints:
+
+  ``` yaml
+    [MetricsMessageQueue]
+    Protocol = 'tcp'
+    Host = '*'
+    Port = 5573
+    Type = 'zero'
+    Topic = 'metrics'
+    [MessageQueue.Optional]
+        # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+        # Client Identifiers
+        Username =""
+        Password =""
+        ClientId ="service-abc"
+        # Connection information
+        Qos = "0" # Quality of service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive =  "10" # Seconds (must be 2 or greater)
+        Retained = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+        SkipCertVerify = "false"
+
+    [CPEMessageQueue]
+    Protocol = 'tcp'
+    Host = '*'
+    Port = 5574
+    Type = 'zero'
+    Topic = 'cpe'
+    [MessageQueue.Optional]
+        # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+        # Client Identifiers
+        Username =""
+        Password =""
+        ClientId ="service-abc"
+        # Connection information
+        Qos = "0" # Quality of service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive =  "10" # Seconds (must be 2 or greater)
+        Retained = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+        SkipCertVerify = "false"
+  ```
+
+- Metrics and CPE collected/monitored and reported by a service will be associated with a name key.  That name key will be used in the configuration to turn `on` the metric or CPE and will be used in the model/DTO when reporting on the metric or CPE. 
+- *Should consideration be given to incorporate alternate protocols/standards for metric and CPE such as https://opentelemetry.io/ or https://github.com/statsd/?*
+- *Should we provide a standard interface for the function that gets called to create a new CPE*
+- *Should we provide a standard interface for the function that gets called by a schedule service (internal or external) to fetch and publish the desired metrics at the appointed time?*
+  - *Do we dictate the use of the scheduler service for this or use the internal scheduler approach?*
+
+## Decision
+
+*To be determined*
+
+## Consequences
+
+- Given the potential that each service publish metrics or CPE, 0MQ is an unlikely implementation unless each service uses a different 0MQ pipe (0MQ topics do not allow multiple publishers).  *Given this constraint, should 0MQ implementation be allowed*

```

Comment:
I don't prefer to use the brokerless protocol like 0MQ, because the subscribers need to know each service's address to listen to the messages.

Reply:
No, ZMQ is not an option and should be stated as so.

Reply:
What if all you wanted to do was get data from one service on a few limited metrics?  A big broker for that at the edge seems a little overkill.  However, I am willing to bend to group consensus on this without other input.